src/variant_detection/variant_detection.cpp

```diff
@@ -4,28 +4,94 @@
 #include <seqan3/io/sam_file/input.hpp>         // SAM/BAM support (seqan3::sam_file_input)
 
 #include "modules/sv_detection_methods/analyze_cigar_method.hpp"    // for the split read method
+#include "modules/sv_detection_methods/analyze_read_pair_method.hpp"// for the read pair method
 #include "modules/sv_detection_methods/analyze_sa_tag_method.hpp"   // for the cigar string method
 #include "variant_detection/bam_functions.hpp"                      // for hasFlag* functions
 
 using seqan3::operator""_tag;
 
+void detect_junctions_in_short_reads_sam_file(std::vector<Junction> & junctions,
+                                              std::filesystem::path const & alignment_short_reads_file_path,
+                                              std::vector<detection_methods> const & methods,
+                                              uint64_t const min_var_length)
+{
+    // Open input alignment file
+    using my_fields = seqan3::fields<seqan3::field::flag,       // 2: FLAG
+                                     seqan3::field::ref_id,     // 3: RNAME
+                                     seqan3::field::ref_offset, // 4: POS
+                                     seqan3::field::mapq,       // 5: MAPQ
+                                     seqan3::field::header_ptr>;
+
+    seqan3::sam_file_input alignment_short_reads_file{alignment_short_reads_file_path, my_fields{}};
+
+    // Check that the file is sorted before proceeding.
+    if (alignment_short_reads_file.header().sorting != "coordinate")
+    {
+        throw seqan3::format_error{"ERROR: Input file must be sorted by coordinate (e.g. samtools sort)"};
+    }
+    uint16_t num_good = 0;
+
+    for (auto & record : alignment_short_reads_file)
+    {
+        seqan3::sam_flag const flag         = record.flag();                            // 2: FLAG
+        int32_t const ref_id                = record.reference_id().value_or(-1);       // 3: RNAME
+        int32_t const ref_pos               = record.reference_position().value_or(-1); // 4: POS
+        uint8_t const mapq                  = record.mapping_quality();                 // 5: MAPQ
+        auto const header_ptr               = record.header_ptr();
+        auto const ref_ids = header_ptr->ref_ids();
+
+        // TODO (irallia): check for mapping quality (e.g. vaquita: see minMapQual)
```

Comment:
Is this TODO outdated, because you check for `mapq < 20` in the next if clause ?